src/context.ts

```diff
@@ -26,18 +27,16 @@ export interface OperonContext {
   readonly span: Span;
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  getConfig(key: string): any;
+  getConfig<T>(key: string, defaultValue?: T): T | undefined;
```

Comment:
```suggestion
  getConfig<T>(key: string): T | undefined;
  getConfig<T>(key: string, defaultValue: T): T;
```

Reply:
this way, the type system knows you can get undefined if you don't pass a default value and you can't get undefined if you do pass a default value